analytic_hr_department_restriction/models/account_analytic_plan.py

```diff
@@ -2,13 +2,23 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
 from odoo import fields, models
+from odoo.tools import config
 
 
 class AccountAnalyticPlan(models.Model):
     _inherit = "account.analytic.plan"
 
-    department_id = fields.Many2one(
-        comodel_name="hr.department",
-        string="Department",
-        domain="['|',('company_id', '=?', company_id),('company_id', '=', False)]",
-    )
+    department_id = fields.Many2one(comodel_name="hr.department", string="Department")
```

Comment:
Why removing the domain?

Reply:
The company field no longer exists in the plans since v17.

Reply:
Yeah, but this is `hr.department`, which has company, so this should be converted to company dependent or be a many2many.

Reply:
Ok, change to many2many.